ipa-core/src/protocol/ipa_prf/prf_eval.rs

```diff
@@ -147,8 +147,9 @@ where
     .await?;
 
     //compute R^(1/z) to u64
-    Ok(zip(gr, z)
-        .map(|(gr, z)| u64::from(gr * z.invert()))
+    let inv_z = crate::ff::ec_prime_field::batch_invert::<N>(&z);
```

Comment:
A nit, that isn't worth changing: I would import this rather than reference via the full path.